lib/cartopy/tests/mpl/test_mpl_integration.py

```diff
@@ -130,6 +130,37 @@ def test_global_scatter_wrap_no_transform():
     plt.scatter(x, y, c=data)
 
 
+@pytest.mark.natural_earth
+@ImageTesting(['global_hexbin_wrap'],
+              tolerance=12.4 if MPL_VERSION < '2.1.0' else 0.5)
+def test_global_hexbin_wrap_new_transform():
+    ax = plt.axes(projection=ccrs.PlateCarree())
+    # By default the coastline feature will be drawn after patches.
+    # By setting zorder we can ensure our hexbin polygons are drawn
+    # after the coastlines.
+    ax.coastlines(zorder=0)
+    x, y = np.meshgrid(np.linspace(-180, 180, 200), np.linspace(-90, 90, 200))
+    data = np.sin(np.sqrt(x**2 + y**2))
+    plt.hexbin(
+        x.flatten(),
+        y.flatten(),
+        C=data.flatten(),
+        transform=ccrs.PlateCarree(),
+        alpha=0.2,
```

Comment:
Could you also lower the bin size so we can see these are hexes on the global map still? The default of 100 seems quite large to me.

Reply:
I am unsure how to do this as this exposes the problem with how matplotlib computes a grid size, if I reduce the points, only whitespace appears between the hexagons like so:

![result-global_hexbin_wrap](https://user-images.githubusercontent.com/210858/80973656-4ac9fc80-8de5-11ea-85b3-4fc71a11aa9d.png)

Reply:
I think I caused a mixup in terms. I meant the gridsize argument to MPL, not the gridsize of the numpy arrays. So, adding an extra argument to your call list: `gridsize=20,` produces something like this for me (still keeping your 200 x 200 numpy size array).

![test_884](https://user-images.githubusercontent.com/12417828/80990646-e95d5a00-8df3-11ea-9612-9a470bc0abb2.png)

Reply:
Hmmm, I don't understand this plot, it shows 20 hexagons in the horizontal, but there should be one hexagon per data point right?  Maybe I am way over my head here with what matplotlib hexbin is doing :(  Maybe the issue is that dx != dy in my example, more boggling.  Thanks for all your effort here!

Reply:
No problem at all. I'm glad you're sticking with it! The hexbin plot is binning the data like a 2d histogram. So, each of these hexagons is summing all of the weighted data within each region, so if you applied no data argument then it would be the sum of the number of points within each hex. The gridsize parameter then tells MPL to fit 20 hexagons in the horizontal and however many are needed vertically to keep them regular I believe.

Reply:
Thanks for your patience with me, I think I have `Voroni` polygons in my head here with these hexbins, which they are not :(